### DISCUSSION TITLE
How does MiXCR recognize CDR3 and by which conserved amino acids?

### DISCUSSION THREAD

**Discussion by linxianzu1995** [milaboratory/mixcr#843] [Q&A] [answered]
# IMGT
A JUNCTION begins at the second **conserved cysteine** of the V-REGION (**2nd-CYS**), at position 104 according to the IMGT unique numbering (Lefranc et al., 2003b). A JUNCTION ends with the **conserved tryptophan** (J-TRP, for the IGH chains) or the **conserved phenylalanine** (J-PHE, for the IG light chains and the TR chains) at position 118 according to the IMGT unique numbering (Lefranc et al., 2003b)

# issue
Checking the output clonotype results, the starting amino acids recognized for CDR3 are Cyc(C), Gly(G), Asp(D), Arg(R), Ser(S), etc.  Are these the starting amino acids that MiXCR uses to recognize CDR3?

**Comment by PoslavskySV** [comment, marked as answer]
Hi, it is probably related to [this](https://github.com/milaboratory/mixcr/discussions/832) discussion as well. @Alex-Davydov could you elaborate please?

**Reply by Alex-Davydov** [reply]
Hi,
Mixcr reference contains all coordinates of the beginning of frameworks and cdrs for V genes (L1, L2, FR1, CDR1, FR2, CDR2, FR3, CDR3) and J genes (JBegin, FR4Begin and FR4End), so the CDR3 region is defined as an nucleotide sequence between CDR3Begin point in V gene and FR4Begin in J gene. The CDR3Begin is the first nucleotide of the codon which encodes conservative cysteine and FR4Begin is the nucleotide right after the triplet which encodes tryptophan or phenylalanine (mixcr works on nucleotides level). More information about the format of mixcr reference libraries you can find [here](https://docs.milaboratories.com/mixcr/reference/ref-repseqio-json-format/).

**Reply by linxianzu1995** [reply]
Hi, As there is no detailed description of the recognition of CDR3 in the align step, only the algorithm for align is described. can you explain the workflow of MiXCR for CDR3 recognition?
